### DISCUSSION TITLE
undefined reference to `SimCore_Anim_DeleteMsgBuf'

### DISCUSSION THREAD

**Discussion by ajay1606** [IPG-Automotive-UK/Aslan-CarMaker#6] [General]
Hello,

Thanks for the very useful repository. Previously we are able to build and test the Aslan -CarMaker project from this repo.

But currently, the same project results in the following error message. 

```
make -C "src" V=1
  -> run 'make -C "src" V=1'
make: Entering directory '/home/ajay/CMProjects/Aslan-CarMaker/src'
gcc -m64 -fPIC -O3 -DNDEBUG -DLINUX -DLINUX64 -D_GNU_SOURCE -D_FILE_OFFSET_BITS=64 -DCM_NUMVER=100201  -I/opt/ipg/carmaker/linux64-10.2.1/include  -Wall -Wimplicit -Wmissing-prototypes -m64 -fPIC -L/opt/ipg/carmaker/linux64-10.2.1/lib -o CarMaker.linux64 \
	 CM_Main.o CM_Vehicle.o User.o /opt/ipg/carmaker/linux64-10.2.1/lib/libcar.a /opt/ipg/carmaker/linux64-10.2.1/lib/libcarmaker.a /opt/ipg/carmaker/linux64-10.2.1/lib/libipgdriver.a /opt/ipg/carmaker/linux64-10.2.1/lib/libipgroad.a /opt/ipg/carmaker/linux64-10.2.1/lib/libipgtire.a /opt/ipg/carmaker/linux64-10.2.1/lib/libz-linux64.a /opt/ipg/carmaker/linux64-10.2.1/lib/liburiparser-linux64.a -lpthread -ldl /opt/ipg/carmaker/linux64-10.2.1/lib/libusb-loader.a -lm -lrt \
	app_tmp.o
CM_Main.o: In function `App_TestRun_Start':
CM_Main.c:(.text+0x2d6): undefined reference to `SimCore_Anim_DeleteMsgBuf'
collect2: error: ld returned 1 exit status
Makefile:48: recipe for target 'CarMaker.linux64' failed
make: *** [CarMaker.linux64] Error 1
make: Leaving directory '/home/ajay/CMProjects/Aslan-CarMaker/src'
  -> Error


###   Script './build_cmrosif.sh' finished with 1 Error(s)!   ###
```

Any response much appreciated.

Regards,
Ajay

**Comment by bogotm** [comment]
Hello,

SimCore_Anim_DeleteMsgBuf is defined in "SimCore.h" in the CarMaker installation "include" folder. It is included in CM_Main by default. Have you made any modifications to the file?

**Comment by ajay1606** [comment]
Thanks for your response,
I have made no changes to an existing project. It is successfully running with CM 10.2.1 version, Now I have upgraded CM with the 11.0.1 version. 
In CM 11.0.1 am getting error.

**Comment by bogotm** [comment]
Hello,

Major CarMaker updates sometimes change the structure of the project and the contents of the C-code in "src". It may be that this has happened in CarMaker 11. 

Normally opening a CarMaker 10 project in CarMaker 11 will update the project files to CM11. However, when it comes to C code, the files are updated to their unmodified CarMaker 11 templates. 

You will have to update the project to CarMaker 11, manually move the changes made to support Project ASLAN (mainly in User.c and Makefile as far as I remember), and then change the CarMaker version in all config files as described in the Readme of this repo: https://github.com/IPG-Automotive-UK/Aslan-CarMaker#change-carmaker-version-optional

**Reply by ajay1606** [reply]
@bogotm Thanks for your quick response.
